pkg/apis/logging/v1/clusterlogging_types.go

```diff
@@ -52,6 +53,11 @@ type ProxySpec struct {
 type LogStoreSpec struct {
 	Type              LogStoreType `json:"type"`
 	ElasticsearchSpec `json:"elasticsearch,omitempty"`
+	RetentionPolicy   *RetentionPolicySpec `json:"retentionPolicy,omitempty"`
+}
+
+type RetentionPolicySpec struct {
+	MaxAge map[v1alpha1.LogSourceType]elasticsearch.TimeUnit `json:"maxAge,omitempty"`
```

Comment:
I think this needs to be something like:
```
type LogStoreSpec struct {
	Type              LogStoreType `json:"type"`
	ElasticsearchSpec `json:"elasticsearch,omitempty"`
	RetentionPolicies   *RetentionPoliciesSpec `json:"retentionPolicy,omitempty"`
}
type RetentionPoliciesSpec struct {
}
type RetentionPolicySpec Struct {
  MaxAge elasticsearch.TimeUnit `json:"maxAge,omitempty"`
}
```

Reply:
the `RetentionPoliciesSpec` struct will have three fields for each log source type ? `App, Infra, Audit` fields of type `RetentionPolicySpec` ?

Reply:
correct.  Im uncertain if we can leave it as a map for flexability or we have to hardcode the field names